Create a group

yukio11 opened the pull request:
## WHY
Users communicate each other in a group
## WHAT
- Make routing to group
- Make the group controller and the views
- Make a form to create and edit a group

yukio11 committed:
Make groups controller and views

yukio11 committed:
Add a sign_out button in the user edit page

yukio11 committed:
Install pry

To use “binding.pry”

yukio11 committed:
Fix routing and association

yukio11 committed:
Make a link to create a group

yukio11 committed:
Make the create action in groups controller

yukio11 committed:
Delete an useless scss file

tetsu8dayo commented on app/controllers/groups_controller.rb:
```diff
@@ -0,0 +1,15 @@
+  def create
+    @group = Group.create(group_params)
+    redirect_to controller: :chat, action: :index
+  end
```
フラッシュメッセージはこれから実装？
時間かからないからささっとやってしまおう

tetsu8dayo replied:
ついでなんでeditも書いてください

yukio11 replied:
edit&フラッシュメッセージ実装しました

yukio11 replied:
[![https://gyazo.com/0c4f9e64e9f807da2dc921aad8c88b9f](https://i.gyazo.com/0c4f9e64e9f807da2dc921aad8c88b9f.png)](https://gyazo.com/0c4f9e64e9f807da2dc921aad8c88b9f)

tetsu8dayo commented on app/views/chat/index.html.haml:
```diff
@@ -3,7 +3,8 @@
       %h2 #{@user.name}
-      = icon('pencil-square-o', class: 'fa-2x')
+      = link_to (new_group_path method: :get) do
+        %i{class:"fa fa-pencil-square-o fa-2x fa-white"}
```
getは書かなくてもいい

yukio11 replied:
削除しました

tetsu8dayo commented on app/views/groups/edit.html.haml:
```diff
@@ -0,0 +1,3 @@
+.chat-group-form
+  %h1 チャットグループ編集
+  = render "form", group: @group
```
groupオプションなんてあったか？
部分テンプレートでもgroupって変数使ってないみたいだけど

tetsu8dayo replied:
部分テンプレートはお好みで
初めからまとめようとしなくてもいいと僕は思う。

tetsu8dayo replied:
むしろまとめるのに凝り始めて開発が進まないと思うよ。

yukio11 replied:
groupオプションの記述削除しました。
部分テンプレートは今後無理にまとめずにやります！

tetsu8dayo commented on Gemfile:
```diff
@@ -53,7 +53,7 @@ group :development do
   gem 'spring-watcher-listen', '~> 2.0.0'
   gem 'erb2haml'
-
+  gem 'pry'
```
better-errors
binding of caller というgemを入れるといろいろ捗る
binding of caller の方は名前がうろ覚えなので当てにしないでください。

yukio11 replied:
better_errors：エラー画面をいい感じに変更してくれる
binding_of_caller：better_errorsのエラー画面でirbを使える
使えそうなのでインストールしました。
ありがとうございます！

tetsu8dayo commented on app/controllers/application_controller.rb:
```diff
@@ -4,6 +4,9 @@ class ApplicationController < ActionController::Base
+  def after_sign_out_path_for(resource)
+    new_user_session_path
+  end
   protected
```
なんじゃこりゃ
deviseだとデフォの設定でログインフォームに飛ばされた記憶があるけど覚えてない
deviseのwikiを見ればカスタマイズ関連は大体書いてありそうだからそっち読んでください
よくわからないのですがオーバーライドする意味あるの？

yukio11 replied:
サインアウト時にリダイレクトされなかったため追記しました。
リダイレクトされないのは、ルートパスを設定していないことが原因だったので、
ルートパスを設定し、after_sign_put_path_forメソッドを削除しました。

tetsu8dayo replied:
そうですね。

yukio11 committed:
Add flash messages in creating and updating a group

Make a show page for group
Add a flash message in creating a group
Add a flash message in updating a group

yukio11 committed:
fix haml

yukio11 committed:
Fix redirecting

When a user signout, app redirect him to signin page.

yukio11 committed:
Fix haml

Delete useless options

yukio11 committed:
Add gem

gem 'better_errors’:Make the error display better
gem ‘binding_of_caller’:To be able to use rib in the error display with
better_errors

oyaoyaoyana commented on app/controllers/groups_controller.rb:
```diff
@@ -1,15 +1,44 @@
+  end
+
+  def edit
+    set_group
```
20行目,24行目
before_action を使いましょう！！
記事長いけど、一通り目を通してみて〜！
http://ruby-rails.hatenadiary.com/entry/20141129/1417223453

yukio11 replied:
before_action追加しました！
たしかに、以前の書き方はDRYに反していますね...
ありがとうございます！

oyaoyaoyana commented on app/views/groups/show.html.haml:
```diff
@@ -1 +1,2 @@
 = @group.name
+= link_to "Edit", edit_group_path
```
edit_group_pathに引数を持たせていないけど、ページ遷移いけてる？

yukio11 replied:
ページ遷移できています！

oyaoyaoyana commented on app/views/chat/index.html.haml:
```diff
@@ -5,7 +5,7 @@
       = link_to (new_group_path method: :get) do
         %i{class:"fa fa-pencil-square-o fa-2x fa-white"}
-      = link_to (edit_user_registration_path method: :get) do
+      = link_to (edit_user_registration_path) do
```
= link_to (edit_user_registration_path) do
のかっこ必要ある？
 = link_to edit_user_registration_path do
かっこは、複雑なコード箇所の意味のあるまとまりを区切る時に使う感じかな！
使いながら覚えていきましょ〜〜〜〜！！！！

yukio11 replied:
リファレンスを見て書いたらこうなってしまったようです...
消しました！

yukio11 committed:
Add before action

To follow DRY

yukio11 committed:
Fix haml

Delete useless ()

oyaoyaoyana commented on app/controllers/groups_controller.rb:
```diff
@@ -1,4 +1,7 @@
 class GroupsController < ApplicationController
+
+  before_action :set_group, only:[:edit, :update]
```
試しに%記法を使ってみましょう！
http://qiita.com/mogulla3/items/46bb876391be07921743

yukio11 committed:
Remove div in haml

div can be omitted in html

tetsu8dayo commented on app/controllers/groups_controller.rb:
```diff
@@ -0,0 +1,45 @@
+    if @group.save
+      redirect_to group_path(@group), notice:"グループの更新が完了しました。"
+    else
+      flash[:alert] = "グループの更新に失敗しました。"
```
flash.now じゃないと別のページに行った後もフラッシュメッセージが残ると思う

yukio11 replied:
修正しました

tetsu8dayo commented on app/models/group.rb:
```diff
@@ -1,4 +1,6 @@
   has_many :messages
+  has_many :group_users
   has_many :users, through: :group_users
+  accepts_nested_attributes_for :users
```
accepts_nested...いらなくないか

tetsu8dayo replied:
消して動いたらいらないね

yukio11 replied:
消して動いたので、消します

yukio11 committed:
Remove accepts_nested_attributes_for from group model

Work correctly without the method

yukio11 committed:
fix flash messages
